packages/playground/src/api/documents/pipelinesTypes.ts

```diff
@@ -28,7 +28,7 @@ export interface DocumentsSourceFile {
   name?: string[];
   directory?: string[];
   content?: string[];
-  metadata?: StringToStringArrayMap;
+  metadata?: Map<string[]>;
```

Comment:
Maybe avoid the use of `Map` as name since it shadows a [global](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)?

Reply:
Maybe just use `Record<string, string[]>` ?

Reply:
I'll look into if we can use Record, due to backwards capability. And thanks, I forgot about `new Map`!

Reply:
What do you mean by "backwards compatibility" in this case?